.github/workflows/continuous-integration.yml

```diff
@@ -96,17 +82,38 @@ jobs:
           echo $HDF5_DIR
           echo "HDF5_DIR=$HDF5_DIR" >> $GITHUB_ENV
 
-      - name: Download a specific release of PETSc
+      - name: Cache PETSc
+        uses: actions/cache@v4
+        id: cache-petsc
+        env:
+          cache-name: cache-PETSc
+        with:
+          path: "./petsc"
+          key: cache-${{ matrix.os }}-${{ matrix.python-version }}
+
+      - if: steps.cache-petsc.outputs.cache-hit != 'true'
+        name: Download a specific release of PETSc
         run: |
           git clone --depth 1 --branch v3.21.4 https://gitlab.com/petsc/petsc.git
 
-      - name: Install PETSc with complex support, and test it
+      - if: steps.cache-petsc.outputs.cache-hit != 'true'
+        name: Install PETSc with complex support
         working-directory: ./petsc
         run: |
           export PETSC_DIR=$(pwd)
           export PETSC_ARCH=petsc-cmplx
           ./configure --with-scalar-type=complex --with-fortran-bindings=0 --have-numpy=1         
-          make all check
+          make all
+          echo "PETSC_DIR=$PETSC_DIR" > petsc.env
+          echo "PETSC_ARCH=$PETSC_ARCH" >> petsc.env
+
+      # This step is not really necessary and could be combined with PETSc install
+      # step; however it's good to verify if the cached PETSc installation really works!
+      - name: Test PETSc installation
+        working-directory: ./petsc
+        run: |
+          source petsc.env
```

Comment:
Are we sure that this file will always exist? It is created in the previous step, if there is no previous cache.

Reply:
Yes; `./petsc` is the declared cached folder (line 91), so its contents will persist throughout multiple CI runs within a single PR. Basically `petsc.env` will be either in cache or created in the previous step.